src/cli/commands/test/iac-local-execution/parsers/kubernetes-parser.ts

```diff
@@ -21,7 +32,24 @@ export function tryParsingKubernetesFile(
         docId,
       };
     } else {
-      throw new Error('Invalid K8s File!');
+      throw new MissingRequiredFieldsInKubernetesYamlError();
     }
   });
 }
+
+class FailedToParseKubernetesYamlError extends CustomError {
+  constructor(message?: string) {
+    super(message || 'Failed to parse Kubernetes YAML file');
+    this.code = IaCErrorCodes.FailedToParseKubernetesYamlError;
+    this.userMessage = '';
+  }
+}
+class MissingRequiredFieldsInKubernetesYamlError extends CustomError {
+  constructor(message?: string) {
+    super(
+      message || 'Failed to detect Kubernetes file, missing required fields',
+    );
+    this.code = IaCErrorCodes.MissingRequiredFieldsInKubernetesYamlError;
+    this.userMessage = '';
```

Comment:
This probably should be filtered out and ignored further upstream.

```suggestion
    this.userMessage = `We were unable to detect whether the YAML file "${filename}" is a valid Kubernetes file, it is missing the following fields: 'apiVersion', 'kind', 'metadata'`;
```